packages/gatsby/src/utils/webpack-error-utils.ts

```diff
@@ -11,42 +12,91 @@ type Stage = keyof typeof stageCodeToReadableLabel
 type StageLabel = typeof stageCodeToReadableLabel[Stage]
 
 interface ITransformedWebpackError {
-  id: "98123"
+  id: string
   filePath?: string
   location?: {
     start: string
   }
   context: {
     stage: Stage
     stageLabel: StageLabel
-    message?: string
+    sourceMessage?: string
+    [key: string]: string | boolean | undefined
   }
 }
 const transformWebpackError = (
   stage: keyof typeof stageCodeToReadableLabel,
   webpackError: any
 ): ITransformedWebpackError => {
-  return {
-    id: `98123`,
-    filePath: webpackError?.module?.resource,
-    location:
-      webpackError?.module?.resource && webpackError?.error?.loc
-        ? {
-            start: webpackError.error.loc,
-          }
-        : undefined,
-    context: {
-      stage,
-      stageLabel: stageCodeToReadableLabel[stage],
-      message: webpackError?.error?.message || webpackError?.message,
+  const handlers = [
+    {
+      regex: /Can't resolve '(.*?)' in '(.*?)'/m,
+      cb: (match): IMatch => {
+        return {
+          id: `98124`,
+          context: {
+            sourceMessage: match[0],
+            packageName: match[1],
+          },
+        }
+      },
     },
+  ]
 
-    // We use original error to display stack trace for the most part.
-    // In case of webpack error stack will include internals of webpack
-    // or one of loaders (for example babel-loader) and doesn't provide
-    // much value to user, so it's purposely omitted.
-    // error: webpackError?.error || webpackError,
+  // We use original error to display stack trace for the most part.
+  // In case of webpack error stack will include internals of webpack
+  // or one of loaders (for example babel-loader) and doesn't provide
+  // much value to user, so it's purposely omitted.
+  // error: webpackError?.error || webpackError,
```

Comment:
Above comment was related to commented out `error` field on structured error we report. After shuffling things around it is bit out of place, but I also don't see great place to put this in given that we have 2 different places we construct error objects